File: test/py/test_registerzadok.py
'Test generation'

from registerzadok import *
from math import *
import numpy as np



def test_test_wheat1():
    params= registerzadok(
    slopeTSFLN = 0.9,
    calendarCumuls = [ 0.0, 112.330110409888,157.969706915664, 280.570678654207],
    calendarMoments = ["Sowing","Emergence","EndVernalisation","MainShootPlus1Tiller"],
    phase = 2,
    calendarDates = ["21/3/2007","27/3/2007","30/3/2007","5/4/2007"],
    TSFLN = 2.6,
     )
    hasZadokStageChanged_estimated = params[0]
    hasZadokStageChanged_computed = 0
    assert (hasZadokStageChanged_estimated == hasZadokStageChanged_computed)
    currentZadokStage_estimated = params[1]
    currentZadokStage_computed = "MainShootPlus1Tiller"
    assert (currentZadokStage_estimated == currentZadokStage_computed)
    calendarMoments_estimated = params[2]
    calendarMoments_computed = ["Sowing","Emergence","EndVernalisation","MainShootPlus1Tiller"]
    assert np.all(calendarMoments_estimated == calendarMoments_computed)
    calendarDates_estimated = params[3]
    calendarDates_computed = ["21/3/2007", "27/3/2007", "30/3/2007", "5/4/2007"]
    assert np.all(calendarDates_estimated == calendarDates_computed)
    calendarCumuls_estimated = np.around(params[4], 2)
    calendarCumuls_computed = [ 0.0, 112.33,157.97, 280.57]
    assert np.all(calendarCumuls_estimated == calendarCumuls_computed)

File: test/java/test_RegisterZadok.java
//Test generation'
package pheno2;
import java.text.ParseException;
import java.util.Arrays;
import java.util.*;
import java.util.List;
import java.util.ArrayList;
import java.math.BigDecimal;
import java.text.SimpleDateFormat;
class test_RegisterZadok
{
    //check wheat model1);
    //test_wheat1);
    public static void test_wheat1()
    {
        double cumulTT = 354.582294511779;
        double phase = 2;
        double leafNumber = 4.8854219661087575;
        List<String> calendarMoments = new ArrayList<>();
        calendarMoments.addAll(Arrays.asList("Sowing","Emergence","EndVernalisation","MainShootPlus1Tiller"));
        SimpleDateFormat format=new SimpleDateFormat("dd/MM/yyyy");
        List<Date> calendarDates=new ArrayList<>() ;
        try
        {
            calendarDates.addAll(Arrays.asList(format.parse("21/3/2007"),format.parse("27/3/2007"),format.parse("30/3/2007"),format.parse("5/4/2007")));
        }
        catch (ParseException e)
        {
        }
        List<Double> calendarCumuls = new ArrayList<>();
        calendarCumuls.addAll(Arrays.asList( 0.0, 112.330110409888,157.969706915664, 280.570678654207));
        double cumulTTFromZC_65 = 0;
        Date currentdate=null ;
        try
        {
        currentdate = format.parse("9/4/2007");
        }
        catch (ParseException e)
        {
        }
        double Der = 300.0;
        double slopeTSFLN = 0.9;
        double intTSFLN = 2.6;
        double FinalLeafNumber = 8.797582013199484;
        String currentZadokStage = "MainShootPlus1Tiller";
        int hasZadokStageChanged = 0;

        RegisterZadok res0 = Estimation_RegisterZadok.CalculateRegisterZadok(cumulTT,phase,leafNumber,calendarMoments,calendarDates,calendarCumuls,cumulTTFromZC_65,currentdate,Der,slopeTSFLN,intTSFLN,FinalLeafNumber,currentZadokStage,hasZadokStageChanged);
        System.out.println(" hasZadokStageChanged: "+res0.hasZadokStageChanged+" currentZadokStage: "+res0.currentZadokStage+" calendarMoments: "+res0.calendarMoments+" calendarDates: "+res0.calendarDates+" calendarCumuls: "+res0.calendarCumuls);

        System.out.println(res0.hasZadokStageChanged==0);
        
        List<Date> calendarDates_calc=null ;
        try
        {
            calendarDates_calc = Arrays.asList(format.parse("21/3/2007"),format.parse("27/3/2007"),format.parse("30/3/2007"),format.parse("5/4/2007"));
        }
        catch (ParseException e)
        {
        }        
        System.out.println((res0.calendarDates).equals(calendarDates_calc));
        List<Double> calendarCumuls_est = new ArrayList<>();
        for (int i=0; i<res0.calendarCumuls.size(); i++){
           calendarCumuls_est .add((new BigDecimal(res0.calendarCumuls.get(i)).setScale(2, BigDecimal.ROUND_HALF_DOWN)).doubleValue());
        }
        System.out.println(calendarCumuls_est.equals( Arrays.asList(0.0, 112.33,157.97, 280.57)));
        System.out.println((res0.calendarMoments).equals(Arrays.asList("Sowing","Emergence","EndVernalisation","MainShootPlus1Tiller")));
        System.out.println((res0.currentZadokStage).equals("MainShootPlus1Tiller"));
    }
}
